driver/heartbeat/heartbeat.cpp

```diff
@@ -46,6 +46,7 @@ freighter::Error heartbeat::Heartbeat::stop() {
     running.wait(false);
     running = false;
     run_thread.join();
+    breaker.close();
```

Comment:
Why do we put this after joining the run thread? Is there a way we can do all of this without needing to manage multiple things i.e. can we get the breaker to exit when running is set to false

Reply:
refactored to close internally on destruction